cli/cook/subcommands/submit.py

```diff
@@ -156,6 +161,16 @@ def submit(clusters, args, _):
     application_version = job.pop('application-version', version.VERSION)
     job['application'] = {'name': application_name, 'version': application_version}
 
+    group = None
+    if 'group-name' in job:
```

Comment:
I thought this code might generate a unique group UUID for each submitted job rather than using the same one for all of them. Your (offline) explanation made sense, but renaming `job` to `job_template` here might make this more clear.

Reply:
Good call, renamed